include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -184,6 +238,13 @@ inline device_memory_resource* get_per_device_resource(cuda_device_id device_id)
 inline device_memory_resource* set_per_device_resource(cuda_device_id device_id,
                                                        device_memory_resource* new_mr)
 {
+  // Note: even though set_per_device_resource() and set_per_device_resource_ref() are not
+  // interchangeable, we call the latter from the former to maintain resource_ref
+  // state consistent with the resource pointer state. This is necessary because the
+  // Python API still uses the raw pointer API. Once the Python API is updated to use
```

Comment:
cudf JNI also uses `set_per_device_resource` and `get_current_device_resource` today, so we'll also be needing to change our code to use `_ref`.

Reply:
🙏